lib/chef_metal_fog/providers/openstack.rb

```diff
@@ -32,6 +32,45 @@ def self.compute_options_for(provider, id, config)
         [result, id]
       end
 
+      # Check given IP already attached and attach if necessary
+      def attach_floating_ips(action_handler, machine_spec, machine_options, server)
+        if option_for(machine_options, :floating_ip_pool)
+          Chef::Log.info 'Attaching IP from pool'
+          action_handler.perform_action "attach floating IP from #{option_for(machine_options, :floating_ip_pool)} pool" do
+            pool = option_for(machine_options, :floating_ip_pool)
```

Comment:
Since you reference this multiple times, you can move it to the top of the method definition.